app/uk/gov/hmrc/cgtpropertydisposalsfrontend/connectors/CGTPropertyDisposalsConnector.scala

```diff
@@ -30,7 +34,7 @@ import scala.concurrent.{ExecutionContext, Future}
 @ImplementedBy(classOf[CGTPropertyDisposalsConnectorImpl])
 trait CGTPropertyDisposalsConnector {
 
-  def getBusinessPartnerRecord(nino: NINO, name: Name, dob: DateOfBirth)(
+  def getBusinessPartnerRecord(entity: Either[Trust,Individual])(
     implicit hc: HeaderCarrier
```

Comment:
Wouldn't this have been better as a sub-type of a sealed trait? That would allow us to easily extend the code for other types of users e.g. agents.

Reply:
individuals and trusts should be the only two types of entities we get a bpr using register with id for. We wouldn't get a business partner for an agent since we don't allow them to go through a subscription journey - the client needs to go through the journey themselves